lib/friends/introvert.rb

```diff
@@ -777,5 +779,52 @@ def thing_with_name_in(type, text)
     def bad_line(expected, line_num)
       raise FriendsError, "Expected \"#{expected}\" on line #{line_num}"
     end
+
+    def decide_default_location_output(activity)
+      @activities = stable_sort(@activities)
+      existing_activities = @activities - [activity]
+
+      str = "Default location"
+      if activity != @activities.first
+        str += " from #{earliest_consecutive_default_location_activity_date(activity, existing_activities)} to"
+        str += " #{next_activity_date_with_different_default_location(activity, existing_activities) || 'present'}"
+      end
+
+      str += " already" if last_default_location_same_as_added_default_location?(activity, existing_activities)
+      str += " set to: \"#{activity.default_location}\""
+
+      @output << str
+    end
+
+    def last_default_location_same_as_added_default_location?(activity, existing_activities)
+      last_default_location_activity = existing_activities.select { |a| a.default_location && (a.date < activity.date) }.first
+      if last_default_location_activity
+        last_default_location_activity.default_location == activity.default_location
+      end
+    end
+
+    def earliest_consecutive_default_location_activity_date(activity, existing_activities)
+      # binding.pry
+      earlier_default_location_activities = existing_activities.select { |a| a.default_location && (a.date < activity.date) }
+      consecutively_earliest_activity_with_same_default_location = nil
+      earlier_default_location_activities.each do |act|
+        break if act.default_location != activity.default_location
+        consecutively_earliest_activity_with_same_default_location = act if act.default_location == activity.default_location
+      end
+      if consecutively_earliest_activity_with_same_default_location
```

Comment:
In Ruby, the last line of a method is the method's return value, and whichever branch of a conditional gets evaluated is what the conditional evaluates to (`nil` if there's no `else` branch and none of the conditional branches are evaluated). So this could just be:

```ruby
if consecutively_earliest_activity_with_same_default_location
  consecutively_earliest_activity_with_same_default_location.date
else
  activity.date
end
```

or

```ruby
if consecutively_earliest_activity_with_same_default_location
  return consecutively_earliest_activity_with_same_default_location.date
end

activity.date
```

or

```ruby
(consecutively_earliest_activity_with_same_default_location || activity).date
```

Reply:
Ooo I like the last line, much shorter and more readable 😍 

I've seen people use `return` in other apps, I think it is to be super explicit with what is happening in the code. I guess if the code is a little busier, then there is a case for using it? Otherwise (like in this method), the return value is obvious 👍